pkg/collector/corechecks/cluster/orchestrator/manifest_buffer.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && orchestrator
+// +build kubeapiserver,orchestrator
+
+package orchestrator
+
+import (
+	"time"
+
+	"go.uber.org/atomic"
+
+	model "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ManifestBufferConfig contains information about buffering manifests.
+type ManifestBufferConfig struct {
+	KubeClusterName             string
+	ClusterID                   string
+	MaxPerMessage               int
+	MaxWeightPerMessageBytes    int
+	MsgGroupRef                 *atomic.Int32
+	BufferedManifestEnabled     bool
+	MaxBufferedManifests        int
+	ManifestBufferFlushInterval time.Duration
+}
+
+// ManifestBuffer is a buffer of manifest sent from all collectors
+// It has a slice bufferedManifests used to buffer manifest and stop channel
+// ManifestBuffer is started as a dedicated thread each time CollectorBundle runs a check
+// and gets stopped after the check is done.
+type ManifestBuffer struct {
+	Cfg               *ManifestBufferConfig
+	ManifestChan      chan interface{}
+	bufferedManifests []interface{}
+	stopCh            chan struct{}
+}
+
+// NewManifestBuffer returns a new ManifestBuffer
+func NewManifestBuffer(chk *OrchestratorCheck) *ManifestBuffer {
+	manifestBuffer := &ManifestBuffer{
+		Cfg: &ManifestBufferConfig{
+			ClusterID:                   chk.clusterID,
+			KubeClusterName:             chk.orchestratorConfig.KubeClusterName,
+			MsgGroupRef:                 chk.groupID,
+			MaxPerMessage:               chk.orchestratorConfig.MaxPerMessage,
+			MaxWeightPerMessageBytes:    chk.orchestratorConfig.MaxWeightPerMessageBytes,
+			BufferedManifestEnabled:     chk.orchestratorConfig.BufferedManifestEnabled,
+			MaxBufferedManifests:        chk.orchestratorConfig.MaxPerMessage,
+			ManifestBufferFlushInterval: chk.orchestratorConfig.ManifestBufferFlushInterval,
+		},
+		ManifestChan: make(chan interface{}),
+		stopCh:       make(chan struct{}),
+	}
+	manifestBuffer.bufferedManifests = make([]interface{}, 0, manifestBuffer.Cfg.MaxBufferedManifests)
+
+	return manifestBuffer
+}
+
+// flushManifest flushes manifests by chunking them first then sending them to the sender
+func (cb *ManifestBuffer) flushManifest(sender aggregator.Sender) {
+	manifests := cb.bufferedManifests
+	cb.bufferedManifests = cb.bufferedManifests[:0]
```

Comment:
it is fine here, but should we be safe and reset it at the end (after sending)? 
QQ: This won't run concurrently /multiple workers for the same manifestBuffer, right? As the manifestBuffer would be shared what we probably don't want

Reply:
yeah, I can reset it at the end for safety.

We don't have multiple workers access to the same manifestBuffer. That's why I didn't use `sync.Mutex` here